hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/InventoryServiceIspn.java

```diff
@@ -291,4 +284,53 @@ private boolean isEmpty(Collection c) {
         return c == null || c.isEmpty();
     }
 
+    @Override
+    public void buildExport(OutputStream os) throws IOException {
+        try (Writer writer = new BufferedWriter(new OutputStreamWriter(os))) {
+            writer.write("{\"types\":[\n");
```

Comment:
@lucasponce do you see a better way to do it, than string-writing json? I'm stuck with StreamingOutput + paginated combination, I cannot jsonify with Jackson a yet-incomplete object. You think it's ok?

Reply:
I know that jackson offers some streaming cappabilities, but honestly I haven't tested them yet, so only I read API and examples. Perhaps is worth to take a look and investigate this together.
I remember that in alerting the use was different as the format was not a json per se, but chunks of jsons, but on this case it seems we could rely on a library as the output is a fully json object.

Reply:
I remember I ran this example locally
http://www.baeldung.com/jackson-streaming-api
but I didn't investigate with detail if I can flush in the intermediate, to check that definetly we can control the buffer and avoid to have all data in memory.

Reply:
Thanks, I've already seen it but it went out of my head :)